app/src/androidTest/java/com/epfl/drawyourpath/database/FirebaseDatabaseUtilsTest.kt

```diff
@@ -263,6 +354,93 @@ class FirebaseDatabaseUtilsTest {
+    }
+
+    @Test
+    fun transformLocalDateTimeReturnsNullWhenDataSnapshotContainsNullValues() {
+        assertEquals(FirebaseDatabaseUtils.transformLocalDateTime(null), null)
+    }
+
+    @Test
+    fun transformPostReturnsExpectedData() {
+        val post = TournamentPost(
+            userId = "user1",
+            run = Run(
+                Path(listOf(listOf(LatLng(0.1, 1.0), LatLng(1.1, 0.2)))),
+                10,
+                10,
+                20,
+                "Lion",
+                0.8,
+            ),
+            votes = 2,
+            date = LocalDateTime.now().minusDays(2),
+            usersVotes = mutableMapOf("user1" to 1, "user2" to 1),
+        )
+        val postSnapshot = mockTournamentPost(post)
+        val transformedPost = FirebaseDatabaseUtils.transformPost(postSnapshot)
+        assertThat(transformedPost!!.userId, `is`(post.userId))
+        // Just to know if the run is the same one
+        assertThat(transformedPost.run.getCalories(), `is`(post.run.getCalories()))
+        assertThat(transformedPost.getVotes(), `is`(post.getVotes()))
+        assertThat(transformedPost.date, `is`(post.date))
+        assertThat(transformedPost.getUsersVotes(), `is`(post.getUsersVotes()))
+    }
+
+    @Test
+    fun transformPostReturnsNullWhenNullSnapshot() {
+        assertEquals(FirebaseDatabaseUtils.transformPost(null), null)
+    }
+
+    @Test
+    fun transformPostListReturnsExpectedData() {
+        val posts = listOf(
+            TournamentPost(
+                userId = "user1",
+                run = Run(
+                    Path(listOf(listOf(LatLng(0.1, 1.0), LatLng(1.1, 0.2)))),
+                    10,
+                    10,
+                    20,
+                    "Lion",
+                    0.8,
+                ),
+                votes = 2,
+                date = LocalDateTime.now().minusDays(2),
+                usersVotes = mutableMapOf("user1" to 1, "user2" to 1),
+            ),
+            TournamentPost(
+                "user2",
+                run = Run(
+                    Path(listOf(listOf(LatLng(0.1, 1.0), LatLng(1.1, 0.2)), listOf(LatLng(9.9, 10.11)))),
+                    30,
+                    20,
+                    50,
+                ),
+                votes = 0,
+                date = LocalDateTime.now().minusDays(3),
+                usersVotes = mutableMapOf("user1" to -1, "user2" to 1),
+            ),
+        )
+        val postsSnapshot = posts.map { mockTournamentPost(it) }
+        val snapshot = mock(DataSnapshot::class.java)
+        `when`(snapshot.children).thenReturn(postsSnapshot)
+        val transformedPosts = FirebaseDatabaseUtils.transformPostList(snapshot)
+
+        assertThat(transformedPosts.size, `is`(posts.size))
+    }
+
+    @Test
+    fun transformPostListReturnsEmptyListWithNullSnapshot() {
+        assertEquals(FirebaseDatabaseUtils.transformPostList(null), emptyList<TournamentPost>())
+    }
+
```

Comment:
Suggestion: It would be nice to add a few more comments to improve code readability

Reply:
I really don't see where I could add comments that would improve the code readability here... maybe at line 431 ? It's a very simple mock so I don't really see the benefit.